pages/src/collections/PageCollectionConfig.ts

```diff
@@ -93,6 +94,10 @@ export const createPageCollectionConfig = ({
         ...(incomingCollectionConfig.hooks?.afterChange || []),
         setVirtualFieldsAfterChange,
       ],
+      beforeDelete: [
+        ...(incomingCollectionConfig.hooks?.beforeDelete || []),
+        preventParentDeletion,
```

Comment:
Let's make this opt-outable. Could you please add a 'preventParentDeletion' field (defaults to `true`) to the plugin config and only inject the hook if its `true`?